src/main/java/net/fabricmc/api/settings/JanksonSettings.java

```diff
@@ -0,0 +1,91 @@
+package net.fabricmc.api.settings;
+
+import blue.endless.jankson.Jankson;
+import blue.endless.jankson.JsonElement;
+import blue.endless.jankson.JsonObject;
+import blue.endless.jankson.JsonPrimitive;
+import blue.endless.jankson.impl.SyntaxError;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Map;
+
+public class JanksonSettings extends Settings<JsonElement> {
+
+	public JanksonSettings(String name) {
+		super(name);
+	}
+
+	public JanksonSettings() {
+		super();
+	}
+
+	@Override
+	public void deserialize(InputStream stream, boolean compressed) throws IOException {
+		Jankson jankson = Jankson.builder().build();
+		try {
+			JsonElement element = jankson.load(stream);
```

Comment:
The return type here is JsonObject, you could save yourself some checks and headaches by using it here.

Reply:
Is that always the case? What if we're loading a file containing only `["A", "B"]`? Wouldn't that return a `JsonArray`?

Reply:
Yes, that is always the case. The file
```json5
["A", "B"]
```
should return a SyntaxError with the message, "Found character '[' instead of '{' while looking for the start of an object", because the "root context" of `load` is a JsonObject. If you want to load a file with the possibility of the root being a non-Object element, you can use `loadElement`.